feature/permissions/src/main/java/com/google/jetpackcamera/permissions/PermissionsViewModel.kt

```diff
@@ -80,21 +81,44 @@ class PermissionsViewModel @AssistedInject constructor(
  * - optional permissions that have not yet been denied by the user
  */
 @OptIn(ExperimentalPermissionsApi::class)
-fun getRequestablePermissions(
-    permissionStates: MultiplePermissionsState
-): MutableSet<PermissionEnum> {
-    val unGrantedPermissions = mutableSetOf<PermissionEnum>()
-    for (permission in permissionStates.permissions) {
-        // camera is always required
-        if (!permission.status.isGranted && permission.permission == Manifest.permission.CAMERA) {
-            unGrantedPermissions.add(PermissionEnum.CAMERA)
-        }
-        // audio is optional
-        else if ((!permission.status.shouldShowRationale && !permission.status.isGranted) &&
-            permission.permission ==
-            Manifest.permission.RECORD_AUDIO
-        ) {
-            unGrantedPermissions.add(PermissionEnum.RECORD_AUDIO)
+fun getRequestablePermissions(permissionStates: MultiplePermissionsState): Set<PermissionEnum> {
+    val unGrantedPermissions = buildSet {
+        permissionStates.permissions.forEach { permissionState ->
+            when (permissionState.permission) {
+                // camera is always required
+                Manifest.permission.CAMERA -> {
+                    if (!permissionState.status.isGranted) {
+                        add(PermissionEnum.CAMERA)
+                    }
+                }
+
+                // optional permissions
+                Manifest.permission.RECORD_AUDIO -> {
+                    if (!permissionState.status.shouldShowRationale &&
+                        !permissionState.status.isGranted
+                    ) {
+                        add(PermissionEnum.RECORD_AUDIO)
+                    }
+                }
+
+                Manifest.permission.READ_EXTERNAL_STORAGE -> {
+                    if (!permissionState.status.shouldShowRationale &&
+                        !permissionState.status.isGranted &&
+                        Build.VERSION.SDK_INT <= Build.VERSION_CODES.P
+                    ) {
+                        add(PermissionEnum.READ_STORAGE)
+                    }
+                }
```

Comment:
Since `WRITE_EXTERNAL_STORAGE` includes `READ_EXTERNAL_STORAGE`, let's just ask for the `WRITE_EXTERNAL_STORAGE` permission only. It is unlikely users would grant a read-only permission in a camera app meant to take pictures.